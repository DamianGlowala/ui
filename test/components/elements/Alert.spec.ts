import { describe, it, expect } from 'vitest'
import { UAlert } from '#components'
import type { TypeOf } from 'zod'
import ComponentRender from '../component-render'

describe('Alert', () => {
  it.each([
    ['basic case', {}],
    ['with tile', { props: { title: 'Alert Title' } }],
    ['with title slot', { slots: { title: 'Alert Title slot' } }],
    ['with description', { props: { title: 'Alert Title', description: 'This is an alert description.' } }],
    ['with description slot', { slots: { title: 'Alert Title', description: 'This is an alert description slot' } }],
    ['with icon', { props: { title: 'Alert Title', icon: 'heroicons-information-circle', ui: { icon: { base: 'w-5 h-5' } } } }],
    ['with avatar', { props: { title: 'Alert Title', avatar: { src: 'avatar.jpg', alt: 'User Avatar' }, ui: { avatar: { base: 'flex-shrink-0 self-center' } } } }],
    ['with close button', { props: { title: 'Alert Title', closeButton: { label: 'Close' }, ui: { default: { closeButton: '' } } } }],
    ['with actions', { props: { title: 'Alert Title', actions: [() => { }], ui: { default: { actionButton: { size: 'sm', color: 'red', variant: 'outline' } } } } }],
    ['with custom color', { props: { title: 'Alert Title', color: 'red' } }],
    ['with custom variant', { props: { title: 'Alert Title', variant: 'outline' } }],
    // @ts-ignore
  ])('renders %s correctly', async (nameOrHtml: string, options: TypeOf<typeof Alert.props>) => {
    if (options !== undefined) {
      options.slots = options.slots || { default: () => 'label' }
      options.slots.default = options.slots.default || (() => 'label')
    }
    const html = await ComponentRender(nameOrHtml, options, UAlert)
    expect(html).toMatchSnapshot()
  })
})
