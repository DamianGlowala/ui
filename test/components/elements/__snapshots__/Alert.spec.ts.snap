// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Alert > renders basic case correctly 1`] = `
"<div class="w-full relative overflow-hidden rounded-lg p-4 text-gray-900 dark:text-white bg-white dark:bg-gray-900 ring-1 ring-gray-200 dark:ring-gray-800">
  <div class="flex items-center">
    <!--v-if-->
    <!--v-if-->
    <div class="">
      <!--v-if-->
      <!--v-if-->
      <!--v-if-->
    </div>
    <!--v-if-->
  </div>
</div>"
`;

exports[`Alert > renders with custom color correctly 1`] = `
"<div class="w-full relative overflow-hidden rounded-lg p-4 bg-red-500 dark:bg-red-400 text-white dark:text-gray-900">
  <div class="flex items-center">
    <!--v-if-->
    <!--v-if-->
    <div class="">
      <p class="">Alert Title</p>
      <!--v-if-->
      <!--v-if-->
    </div>
    <!--v-if-->
  </div>
</div>"
`;

exports[`Alert > renders with custom variant correctly 1`] = `
"<div class="w-full relative overflow-hidden rounded-lg p-4 text-white-500 dark:text-white-400 ring-1 ring-inset ring-white-500 dark:ring-white-400">
  <div class="flex items-center">
    <!--v-if-->
    <!--v-if-->
    <div class="">
      <p class="">Alert Title</p>
      <!--v-if-->
      <!--v-if-->
    </div>
    <!--v-if-->
  </div>
</div>"
`;

exports[`Alert > renders with description slot correctly 1`] = `
"<div class="w-full relative overflow-hidden rounded-lg p-4 text-gray-900 dark:text-white bg-white dark:bg-gray-900 ring-1 ring-gray-200 dark:ring-gray-800">
  <div class="flex items-start">
    <!--v-if-->
    <!--v-if-->
    <div class="">
      <!--v-if-->
      <p class="">Description slot</p>
      <!--v-if-->
    </div>
    <!--v-if-->
  </div>
</div>"
`;

exports[`Alert > renders with title correctly 1`] = `
"<div class="w-full relative overflow-hidden rounded-lg p-4 text-gray-900 dark:text-white bg-white dark:bg-gray-900 ring-1 ring-gray-200 dark:ring-gray-800">
  <div class="flex items-center">
    <!--v-if-->
    <!--v-if-->
    <div class="">
      <p class="">Alert Title</p>
      <!--v-if-->
      <!--v-if-->
    </div>
    <!--v-if-->
  </div>
</div>"
`;

exports[`Alert > renders with title slot correctly 1`] = `
"<div class="w-full relative overflow-hidden rounded-lg p-4 text-gray-900 dark:text-white bg-white dark:bg-gray-900 ring-1 ring-gray-200 dark:ring-gray-800">
  <div class="flex items-center">
    <!--v-if-->
    <!--v-if-->
    <div class="">
      <p class="">Title slot</p>
      <!--v-if-->
      <!--v-if-->
    </div>
    <!--v-if-->
  </div>
</div>"
`;
